getndazn/kopytko-framework
# feat!: Add HTTP request and support HTTP caching

## What did you implement:

- HTTP caching using native cachefs respecting HTTP cache-related headers and ETag header support
- New HttpRequest node extending current base Request task node making using HTTP module much easier



## How did you implement it:



## How can we verify it:

Implement a task node extending a new HttpRequest to send a GET request. Send the request multiple times and notice only one request sent via HTTP(S) until the response's max-age value is not reached. If your service implements the ETag mechanism, notice a cached response returned in case of 304 HTTP response.



## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO (introduced few @deprecated mentions)
